### PR TITLE
Inicial commit Bouncy Castle API

### PR DESCRIPTION
**Pull request by alvarolobato**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ ~ THE SOFTWARE.
+ -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.jenkins-ci.plugins</groupId>
+    <artifactId>plugin</artifactId>
+    <version>2.7</version>
```
Comment (amuniz) [inline pom.xml]: Why not 2.9?

File: src/main/java/jenkins/bouncycastle/api/PEMManager.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+ * Fingerprints, etc The supported algorithms will depend on the underlying version of BouncyCastle
+ *
+ * @author Alvaro Lobato
+ */
+public class PEMManager {
+
+    /**
+     * Stores the internal Bouncy Castle or JCA object
+     */
+    Object pemObject;
```
Comment (amuniz) [inline src/main/java/jenkins/bouncycastle/api/PEMManager.java]: `private`?

File: src/main/java/jenkins/bouncycastle/api/PEMManager.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+     * Creates a {@link PEMManager} from an {@link Object} that can be of any supported type by Bouncy Castle or JCA:
+     * {@link Key}, {@link PrivateKey}, {@link KeyPair}, {@link Certificate}, etc.
+     *
+     * @param pemObject object to manage
+     */
+    public PEMManager(@Nonnull Object pemObject) {
+        this.pemObject = pemObject;
+    }
+
+    protected void decodePEM(@Nonnull File pemFile) throws IOException {
```
Comment (amuniz) [inline src/main/java/jenkins/bouncycastle/api/PEMManager.java]: Is this method designed for extension? If not, better `private`.

File: src/main/java/jenkins/bouncycastle/api/PEMManager.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+    }
+
+    /**
+     * Encodes the current stored information in PEM format and returns it as a {@link String}
+     * 
+     * @return PEM encoded data
+     * @throws IOException launched if a problem exists generating the PEM information
+     */
+    @Nonnull
+    public String encodePEM() throws IOException {
```
Comment (amuniz) [inline src/main/java/jenkins/bouncycastle/api/PEMManager.java]: This covers the use case:

```
Object pemObject = ...;
PEMManager pm = new PEMManager(pemObject);
String pemAsString = pm.encodePEM();
```

right?

What about the reverse?

```
String pemString = "PEM as string...";
PEMManager pm = new PEMManager(pemString);
Object pemObject = pm.decodePEM();
```

Is it not a valid use case for the API?

Comment (alvarolobato) [inline src/main/java/jenkins/bouncycastle/api/PEMManager.java]: Not exactly. We want to abstract all the BouncyCastle classes from the caller. private Object pemObject could be a JCA object like Key or KeyPair, or a BouncyCastle object like PEMKeyPair (this is 1.54 specific), so this is managed internally. When you obtain the information we only return JCA version of that object: Key, KeyPair, Certificate...

On the creation we support receiving an object which could be any of the tree above... actually could also be a BC object, but that is not the intended case, we want to expose only JCA.

**Pull request merged**